### PR TITLE
Prepare 1.2.1

### PR DESCRIPTION
**Pull request by ludovicsteinbach**
Cherry-picked:
- #247 
- #248 
- #249 
- #251 
- #256 

Also bumps the version to 1.2.1

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,21 @@
 # Changelog
```
Comment (da1910) [inline CHANGELOG.md:2]: I imagine we want to list the dependency bump under 1.2.1 here?

**Review by da1910 — CHANGES_REQUESTED** [review]
Changelog needs updating for 1.2.1, otherwise looks good

**Review by da1910 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,5 +1,34 @@
...
 
+## grantami-recordlists 1.2.1, 2024-05-07
+
+### Changed
```
Comment (Andy-Grigg) [inline CHANGELOG.md:5]: Is 'Changed' the right word here? Maybe 'Dependencies'? I'm not sure what the towncrier template uses, but maybe we should standardize on that

**Review by Andy-Grigg — APPROVED** [review]
One minor change, but aside from that all looks good

File: CHANGELOG.md (hunk 2 above)
Comment (Andy-Grigg) [reply CHANGELOG.md:5]: Yea, let's go with Dependencies. At least then it's clear that the only meaningful fix is that change

Comment (ludovicsteinbach) [reply CHANGELOG.md:5]: Any thoughts on whether the other cherry picked PRs ought to be listed there too? I did not include them, based on what was in the changelog for previous versions

**Pull request merged**